source/bringauto/external_client/connection/ExternalConnection.cpp

```diff
@@ -269,7 +269,7 @@ u_int32_t ExternalConnection::getNextStatusCounter() {
 }
 
 void ExternalConnection::endConnection(bool completeDisconnect = false) {
-	state_.exchange(ConnectionState::NOT_CONNECTED);
+	state_.exchange(ConnectionState::NOT_INITIALIZED);
```

Comment:
Is NOT_CONNECTED used? Or not?

Reply:
It's only being checked in one place (line 345), but that can't be reached after the endConnection function is called

Reply:
So what is the meaning of NOT_CONNECTED? What is the difference between NOT_CONNECTED and NOT_INITIALIZED?

Reply:
NOT_CONNECTED means that the connect sequence isn't finished yet
NOT_INITIALIZED needs to be set for MG to start the connect sequence and initialize all connected devices (it will remain in that state until a device connects)

Reply:
But not initialized is used in a context where the server expects the structure is not initialized. So NO INITIALIZED is the initial status of the connection.
The connection is NOT CONNECTED, not NOT_INITIALIZED. So there is a problem somewhere else not in this part of the system. Documentation for each state is clear.

If I end connection the connection is NOT_CONNECTED but the structure is still initialized.

Reply:
endConnection was renamed to deinitializeConnection for clarity